tendermint-rs/src/merkle.rs

```diff
@@ -0,0 +1,116 @@
+//! Merkle tree used in Tendermint networks
+
+use sha2::{Digest, Sha256};
+
+/// Size of Merkle root hash
+pub const HASH_SIZE: usize = 32;
+
+/// Compute a simple Merkle root from the arbitrary sized byte slices
+pub fn simple_hash_from_byte_slices(byte_slices: &[&[u8]]) -> [u8; HASH_SIZE] {
```

Comment:
This might be an interesting place to use an iterator bound, e.g.:

```rust
pub fn compute_simple_hash<'a, T>(byte_slices: T) -> [u8; HASH_SIZE]
where
    T: IntoIterator<Item = &'a [u8]>
{
```

or with the fancy new `impl Trait` syntax:

```rust
pub fn compute_simple_hash<'a>(byte_slices: impl IntoIterator<Item = &'a [u8]>) ) -> [u8; HASH_SIZE] {
```